modules/media_event_dispatcher/tests/src/Unit/Media/MediaSourceInfoAlterEventTest.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Drupal\Tests\media_event_dispatcher\Unit\Media;
+
+use Drupal;
+use Drupal\Core\DependencyInjection\ContainerBuilder;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\hook_event_dispatcher\HookEventDispatcherInterface;
+use Drupal\media_event_dispatcher\Event\Media\MediaSourceInfoAlterEvent;
+use Drupal\Tests\hook_event_dispatcher\Unit\HookEventDispatcherManagerSpy;
+use Drupal\Tests\UnitTestCase;
+use function media_event_dispatcher_media_source_info_alter;
+
+/**
+ * Class MediaSourceInfoAlterEventTest.
+ *
+ * @package Drupal\Tests\media_event_dispatcher\Unit\Media
+ *
+ * @group media_event_dispatcher
+ */
+class MediaSourceInfoAlterEventTest extends UnitTestCase {
+
+  /**
+   * The manager.
+   *
+   * @var \Drupal\Tests\hook_event_dispatcher\Unit\HookEventDispatcherManagerSpy
+   */
+  private $manager;
+
+  /**
+   * Sets up the test.
+   */
+  public function setUp() {
+    $builder = new ContainerBuilder();
+    $this->manager = new HookEventDispatcherManagerSpy();
+    $builder->set('hook_event_dispatcher.manager', $this->manager);
+    $builder->compile();
+    Drupal::setContainer($builder);
+  }
+
+  /**
+   * MediaSourceInfoAlterEvent sources array alter test.
+   *
+   * This tests altering the media source plugin definitions array.
+   */
+  public function testMediaSourceInfoAlter() {
+    $sources = $expectedSources = [
+      'image' => [
+        'id' => 'image',
+        'label' => new TranslatableMarkup('Image'),
+        'description' => new TranslatableMarkup('Use local images for reusable media.'),
+        'class' => 'Drupal\media\Plugin\media\Source\Image',
+        'allowed_field_types' => ['image'],
+      ],
+    ];
+
+    $this->manager->setEventCallbacks([
+      HookEventDispatcherInterface::MEDIA_SOURCE_INFO_ALTER => function (
+        MediaSourceInfoAlterEvent $event
+      ) {
+        $sources = &$event->getSources();
+
+        $sources['image']['allowed_field_types'][] = 'test';
+      },
+    ]);
+
+    // Run the procedural hook which should trigger the above handler.
+    media_event_dispatcher_media_source_info_alter($sources);
+
+    $expectedSources['image']['allowed_field_types'][] = 'test';
+
+    $this->assertSame($expectedSources, $sources);
+
+    /** @var \Drupal\media_event_dispatcher\Event\Media\MediaSourceInfoAlterEvent $event */
+    $event = $this->manager->getRegisteredEvent(
+      HookEventDispatcherInterface::MEDIA_SOURCE_INFO_ALTER
+    );
+
+    $this->assertSame($sources, $event->getSources());
```

Comment:
Nitpick: Preferably check the event first before the outcome so an error would pop up earlier. Try to follow the flow in the application, does that makes sense to you?

Reply:
Do you mean to move lines 74-79 up higher, right after the ```media_event_dispatcher_media_source_info_alter($sources)```, or just to swap the parameters on line 79?

Reply:
Yeah move 75-79 right after the hook call before the expectedSources check vs the sources and check with the expectedSources.